packages/ethereum-storage/src/smart-contract-manager.ts

```diff
@@ -282,16 +289,63 @@ export default class SmartContractManager {
         // When set to true, we use it to ignore next confirmation event function call
         let ethereumMetadataCreated: boolean = false;
 
+        // Keep the transaction hash for future needs
+        let transactionHash: string = '';
+
         this.requestHashSubmitter.methods
           .submitHash(contentHash, feesParametersAsBytes)
           .send({
             from: account,
             gas: '100000',
             gasPrice: gasPriceToUse,
+            nonce,
             value: fee,
           })
-          .on('error', (transactionError: string) => {
-            reject(Error(`Ethereum transaction error:  ${transactionError}`));
+          .on('transactionHash', (hash: any) => {
+            // Store the transaction hash in case we need it in the future
+            transactionHash = hash;
+          })
+          .on('error', async (transactionError: string) => {
+            // If failed because of polling timeout, try to resubmit the transaction with more gas
+            if (
+              transactionError.toString().includes(TRANSACTION_POLLING_TIMEOUT) &&
+              transactionHash
+            ) {
+              // If we didn't set the nonce, find the current transaction nonce
+              if (!nonce) {
+                const tx = await this.eth.getTransaction(transactionHash);
+                nonce = tx.nonce;
+              }
+
+              // Get the new gas price for the transaction
+              const newGasPrice = new bigNumber(
+                await gasPriceDefiner.getGasPrice(StorageTypes.GasPriceType.FAST, this.networkName),
+              );
+
+              // If the new gas price is higher than the previous, resubmit the transaction
+              if (newGasPrice.gt(new bigNumber(gasPriceToUse))) {
```

Comment:
At this stage could we have a fallback to add an arbitrary amount to the gasPriceToUse to force it through quicker? e.g. gasPriceToUse + 20 GWEI. Only useful for extreme cases

Reply:
Here we will get the new `fast` gas price. For example, if the last gas price was 17, but it got stuck because now it's 50, we will retry with 50.
But I think you have a good point, we can maybe use the new fast gas price + something like 20% to compensate for the case our gas price APIs are lagging behind.